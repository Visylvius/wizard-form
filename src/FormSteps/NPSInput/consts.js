const NPS_SCORE = 'npsScore';

export { NPS_SCORE };
