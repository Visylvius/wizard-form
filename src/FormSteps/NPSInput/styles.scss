@import '../../variables';

@mixin nps-hover {
  background-color: $nps-hover;
  color: $plain-white;
}

.NPSInput-container {
  display: flex;
  flex-wrap: wrap;

  input[type='radio'] {
    display: none;
  }

  .NPSInput-radioContainer {
    display: flex; 
    justify-content: space-between;
    width: 100%;
  }

  .NPSInput-label {
    display: flex;
    align-items: center;
    background-color: $nps-background;
    border-radius: 3px;
    cursor: pointer;
    color: $font-main;
    font-size: 12px;
    height: 27px;
    justify-content: center;
    width: 30px;

    &:hover {
      @include nps-hover;
    }
  }

  .isSelected {
    @include nps-hover;
  }

  .NPSInput-ratingLabels {
    color: $nps-label;
    display: flex;
    font-size: 12px;
    margin-top: 10px;
    justify-content: space-between;
    width: 100%;
  }

  .NPSInput-buttonContainer {
    display: flex;
    justify-content: center;
    width: 100%;
  }

  .NPSInput-Button { 
    margin-top: 10px;
  }

  .NPSInput-radioLabel {
    color: $font-main;
    font-size: 12px;
    margin-bottom: 10px;
  }
}