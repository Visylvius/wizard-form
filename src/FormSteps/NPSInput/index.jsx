import React from 'react';
import classNames from 'classnames';

import RadioInput from '../../Components/RadioInput';
import Button from '../../Components/Button';

import { NPS_SCORE } from './consts.js';
import './styles.scss';

const NPSInput = ({ npsValue, onButtonClick, onChange, currentStep }) => {
  if (currentStep !== 1) {
    return null;
  }

  return (
    <div className='NPSInput-container'>
      <label className='NPSInput-radioLabel'>
        How likely are you to recommend Cole Haan to a friend or colleague?
      </label>
      <div className='NPSInput-radioContainer'>
        {['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10'].map(
          (num, idx) => {
            const labelClasses = classNames('NPSInput-label', {
              isSelected: npsValue === num,
            });

            return (
              <div key={num}>
                <label htmlFor={`nps-${num}`} className={labelClasses}>
                  {num}
                </label>
                <RadioInput
                  data-testid={`RadioInput-${idx}`}
                  id={`nps-${num}`}
                  name={NPS_SCORE}
                  value={num}
                  onClick={onChange}
                />
              </div>
            );
          },
        )}
      </div>
      <div className='NPSInput-ratingLabels'>
        <span>Not at all likely</span>
        <span>Extremely likely</span>
      </div>
      <Button
        data-testid='NPSInput-Button'
        disabled={!npsValue}
        buttonClasses='NPSInput-Button'
        onClick={onButtonClick}
        shouldCenter
      >
        Next
      </Button>
    </div>
  );
};

export default NPSInput;
