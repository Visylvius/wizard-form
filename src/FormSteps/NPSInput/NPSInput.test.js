import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import NPSInput from './index';

describe('NPSInput (Unit)', () => {
  const mockOnChange = jest.fn();
  const mockOnClick = jest.fn();
  const renderNPSInput = props =>
    render(
      <NPSInput
        testId='SelectGoal-select'
        currentStep={1}
        onChange={mockOnChange}
        onButtonClick={mockOnClick}
        selectedVal='default'
        {...props}
      />,
    );

  describe('render', () => {
    test('should render properly', () => {
      const tree = renderer.create(
        <NPSInput
          currentStep={1}
          onChange={mockOnChange}
          onButtonClick={mockOnClick}
          selectedVal='default'
        />,
      );

      expect(tree).toMatchSnapshot();
    });

    test('should render null if the step is not 5', () => {
      const tree = renderer.create(
        <NPSInput
          currentStep={2}
          onChange={mockOnChange}
          onButtonClick={mockOnClick}
          selectedVal='default'
        />,
      );

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Intergration', () => {
    test('should properly call onChange', async () => {
      const { getByTestId, getByText } = renderNPSInput();

      fireEvent.click(getByTestId('RadioInput-0'), {
        target: { value: '2' },
      });

      await waitFor(() => expect(mockOnChange).toHaveBeenCalled());
    });

    test('should properly call onClick', async () => {
      const { getByTestId } = renderNPSInput({ npsValue: '2' });

      userEvent.click(getByTestId('NPSInput-Button'));

      await waitFor(() => expect(mockOnClick).toHaveBeenCalled());
    });

    test('should not call onClick when the button is disabled', async () => {
      const { getByTestId } = renderNPSInput();

      userEvent.click(getByTestId('NPSInput-Button'));

      await waitFor(() => expect(mockOnClick).not.toHaveBeenCalled());
    });
  });
});
