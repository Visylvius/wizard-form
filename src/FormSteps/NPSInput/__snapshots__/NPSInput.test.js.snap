// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPSInput (Unit) render should render null if the step is not 5 1`] = `null`;

exports[`NPSInput (Unit) render should render properly 1`] = `
<div
  className="NPSInput-container"
>
  <label
    className="NPSInput-radioLabel"
  >
    How likely are you to recommend Cole Haan to a friend or colleague?
  </label>
  <div
    className="NPSInput-radioContainer"
  >
    <div>
      <label
        className="NPSInput-label"
        htmlFor="nps-0"
      >
        0
      </label>
      <div
        className="RadioInput-container"
      >
        <input
          className="RadioInput-component"
          data-testid="RadioInput-0"
          id="nps-0"
          name="npsScore"
          onClick={[MockFunction]}
          type="radio"
          value="0"
        />
      </div>
    </div>
    <div>
      <label
        className="NPSInput-label"
        htmlFor="nps-1"
      >
        1
      </label>
      <div
        className="RadioInput-container"
      >
        <input
          className="RadioInput-component"
          data-testid="RadioInput-1"
          id="nps-1"
          name="npsScore"
          onClick={[MockFunction]}
          type="radio"
          value="1"
        />
      </div>
    </div>
    <div>
      <label
        className="NPSInput-label"
        htmlFor="nps-2"
      >
        2
      </label>
      <div
        className="RadioInput-container"
      >
        <input
          className="RadioInput-component"
          data-testid="RadioInput-2"
          id="nps-2"
          name="npsScore"
          onClick={[MockFunction]}
          type="radio"
          value="2"
        />
      </div>
    </div>
    <div>
      <label
        className="NPSInput-label"
        htmlFor="nps-3"
      >
        3
      </label>
      <div
        className="RadioInput-container"
      >
        <input
          className="RadioInput-component"
          data-testid="RadioInput-3"
          id="nps-3"
          name="npsScore"
          onClick={[MockFunction]}
          type="radio"
          value="3"
        />
      </div>
    </div>
    <div>
      <label
        className="NPSInput-label"
        htmlFor="nps-4"
      >
        4
      </label>
      <div
        className="RadioInput-container"
      >
        <input
          className="RadioInput-component"
          data-testid="RadioInput-4"
          id="nps-4"
          name="npsScore"
          onClick={[MockFunction]}
          type="radio"
          value="4"
        />
      </div>
    </div>
    <div>
      <label
        className="NPSInput-label"
        htmlFor="nps-5"
      >
        5
      </label>
      <div
        className="RadioInput-container"
      >
        <input
          className="RadioInput-component"
          data-testid="RadioInput-5"
          id="nps-5"
          name="npsScore"
          onClick={[MockFunction]}
          type="radio"
          value="5"
        />
      </div>
    </div>
    <div>
      <label
        className="NPSInput-label"
        htmlFor="nps-6"
      >
        6
      </label>
      <div
        className="RadioInput-container"
      >
        <input
          className="RadioInput-component"
          data-testid="RadioInput-6"
          id="nps-6"
          name="npsScore"
          onClick={[MockFunction]}
          type="radio"
          value="6"
        />
      </div>
    </div>
    <div>
      <label
        className="NPSInput-label"
        htmlFor="nps-7"
      >
        7
      </label>
      <div
        className="RadioInput-container"
      >
        <input
          className="RadioInput-component"
          data-testid="RadioInput-7"
          id="nps-7"
          name="npsScore"
          onClick={[MockFunction]}
          type="radio"
          value="7"
        />
      </div>
    </div>
    <div>
      <label
        className="NPSInput-label"
        htmlFor="nps-8"
      >
        8
      </label>
      <div
        className="RadioInput-container"
      >
        <input
          className="RadioInput-component"
          data-testid="RadioInput-8"
          id="nps-8"
          name="npsScore"
          onClick={[MockFunction]}
          type="radio"
          value="8"
        />
      </div>
    </div>
    <div>
      <label
        className="NPSInput-label"
        htmlFor="nps-9"
      >
        9
      </label>
      <div
        className="RadioInput-container"
      >
        <input
          className="RadioInput-component"
          data-testid="RadioInput-9"
          id="nps-9"
          name="npsScore"
          onClick={[MockFunction]}
          type="radio"
          value="9"
        />
      </div>
    </div>
    <div>
      <label
        className="NPSInput-label"
        htmlFor="nps-10"
      >
        10
      </label>
      <div
        className="RadioInput-container"
      >
        <input
          className="RadioInput-component"
          data-testid="RadioInput-10"
          id="nps-10"
          name="npsScore"
          onClick={[MockFunction]}
          type="radio"
          value="10"
        />
      </div>
    </div>
  </div>
  <div
    className="NPSInput-ratingLabels"
  >
    <span>
      Not at all likely
    </span>
    <span>
      Extremely likely
    </span>
  </div>
  <div
    className="Button-container"
  >
    <button
      className="Button-component NPSInput-Button"
      data-testid="NPSInput-Button"
      disabled={true}
      onClick={[MockFunction]}
    >
      Next
    </button>
  </div>
</div>
`;
